source/extensions/filters/network/rocketmq_proxy/codec.cc

```diff
@@ -37,7 +37,11 @@ RemotingCommandPtr Decoder::decode(Buffer::Instance& buffer, bool& underflow, bo
 
   auto mark = buffer.peekBEInt<uint32_t>();
   uint32_t header_length = adjustHeaderLength(mark);
-  ASSERT(frame_length > header_length);
+  if (frame_length <= header_length) {
+    // There is an error in frame_length
+    has_error = true;
+    return nullptr;
+  }
   buffer.drain(FRAME_HEADER_LENGTH_FIELD_SIZE);
 
   uint32_t body_length = frame_length - 4 - header_length;
```

Comment:
FRAME_HEADER_LENGTH_FIELD_SIZE is equal to 4.

I think the 4 here must match the argument to `buffer.drain()`.  Recommend replacing `4` with `FRAME_HEADER_LENGTH_FIELD_SIZE`.

Reply:
+1

Reply:
> If the expression is negative, then a negative number is assigned to a `uint32_t`. I believe that is undefined behavior.
> 
> In order to prevent that, the assert above could check that `frame_length > FRAME_HEADER_LENGTH_FIELD_SIZE + header_length`.

I modified it to this
```if (frame_length < header_length + FRAME_HEADER_LENGTH_FIELD_SIZE)```
Does it make sense? I used `<` instead of `<=` because in some unit test cases the body_length is exactly `0`.

Reply:
I don't know enough about this protocol to know if a stream with an empty body is allowed.  

@aaron-ai @lizhanhui : Is this allowed?

Reply:
Given the lack of response, and the existing unit test, lets allow a zero length body for now.